provisioning/baremetal_config_test.go

```diff
@@ -41,63 +41,29 @@ func TestValidateManagedProvisioningConfig(t *testing.T) {
 
 	tCases := []struct {
 		name          string
-		spec          metal3iov1alpha1.ProvisioningSpec
+		spec          *metal3iov1alpha1.ProvisioningSpec
 		expectedError bool
 		expectedMode  metal3iov1alpha1.ProvisioningNetwork
 		expectedMsg   string
 	}{
 		{
 			// All fields are specified as they should including the ProvisioningNetwork
-			name: "ValidManaged",
-			spec: metal3iov1alpha1.ProvisioningSpec{
-				ProvisioningInterface:     "ensp0",
-				ProvisioningIP:            "172.30.20.3",
-				ProvisioningNetworkCIDR:   "172.30.20.0/24",
-				ProvisioningDHCPRange:     "172.30.20.11, 172.30.20.101",
-				ProvisioningOSDownloadURL: "http://172.22.0.1/images/rhcos-44.81.202001171431.0-openstack.x86_64.qcow2.gz?sha256=e98f83a2b9d4043719664a2be75fe8134dc6ca1fdbde807996622f8cc7ecd234",
-				ProvisioningNetwork:       "Managed",
-			},
+			name:          "ValidManaged",
+			spec:          managedProvisioning().build(),
 			expectedError: false,
 			expectedMode:  metal3iov1alpha1.ProvisioningNetworkManaged,
 		},
 		{
-			// ProvisioningNetwork is not specified but ProvisioningDHCPExternal is.
-			name: "ImpliedManaged",
-			spec: metal3iov1alpha1.ProvisioningSpec{
-				ProvisioningInterface:     "eth0",
-				ProvisioningIP:            "172.30.20.3",
-				ProvisioningNetworkCIDR:   "172.30.20.0/24",
-				ProvisioningDHCPRange:     "172.30.20.11, 172.30.20.101",
-				ProvisioningOSDownloadURL: "http://172.22.0.1/images/rhcos-44.81.202001171431.0-openstack.x86_64.qcow2.gz?sha256=e98f83a2b9d4043719664a2be75fe8134dc6ca1fdbde807996622f8cc7ecd234",
-				ProvisioningDHCPExternal:  false,
-			},
-			expectedError: false,
-			expectedMode:  metal3iov1alpha1.ProvisioningNetworkManaged,
-		},
-		{
-			// Verifying default behavior where both ProvisioningNetwork and ProvisioningDHCPExternal are not specified.
-			name: "DefaultManaged",
```

Comment:
I removed this test since it was a duplicate of the one below. There's no concept of "not specifying" a boolean field, it's zero-value is false (unless it's a pointer, which in this case it's not).